ui/contexts/metamask-notifications/metamask-notifications.tsx

```diff
@@ -0,0 +1,56 @@
+import React, { createContext, useContext, useEffect, useMemo } from 'react';
+import { useSelector } from 'react-redux';
+import { useListNotifications } from '../../hooks/metamask-notifications/useNotifications';
+import { selectIsProfileSyncingEnabled } from '../../selectors/metamask-notifications/profile-syncing';
+import { selectIsMetamaskNotificationsEnabled } from '../../selectors/metamask-notifications/metamask-notifications';
+import type { Notification } from '../../../app/scripts/controllers/metamask-notifications/types/notification/notification';
+
+type MetamaskNotificationsContextType = {
+  listNotifications: () => void;
+  notificationsData?: Notification[];
+  isLoading: boolean;
+  error?: unknown;
+};
+
+const MetamaskNotificationsContext = createContext<
+  MetamaskNotificationsContextType | undefined
+>(undefined);
+
+export const useMetamaskNotificationsContext = () => {
+  const context = useContext(MetamaskNotificationsContext);
+  if (!context) {
+    throw new Error(
+      'useNotificationsContext must be used within a MetamaskNotificationsProvider',
+    );
+  }
+  return context;
+};
+
+export const MetamaskNotificationsProvider: React.FC = ({ children }) => {
```

Comment:
Hmm since we are already using redux, I'm not sure about this Context/Provider.

At most this could just be a hook.

Reply:
Investigate, but we don't need to refactor in this PR.

Reply:
Ticket - https://consensyssoftware.atlassian.net/browse/NOTIFY-659